src/frontend/src/utils/gantt.utils.ts

```diff
@@ -37,56 +55,168 @@ export interface GanttTaskData {
   prevStart: Date;
   prevEnd: Date;
   newStart: Date;
   newEnd: Date;
-  duration: number;
-  stage?: WorkPackageStage;
-  teamName: string;
-  baseWbs: WbsNumber;
-  workPackageChanges: RequestEventChange[];
-  createProject?: boolean;
+  type: 'create-project' | 'create-work-package' | 'edit-work-package';
 };
 
-export const applyChangeToEvent = (eventChanges: EventChange[], task: GanttTaskData): GanttTaskData => {
-  let workPackages: GanttTaskData[] = [];
-  if (task && task.workPackages) {
-    workPackages = task.workPackages.map((wpEvent) => applyChangeToEvent(eventChanges, wpEvent));
-  }
+export const getProjectStartDate = (project: ProjectPreview): Date => {
+  return project.workPackages.reduce((acc, current) => {
+    if (current.startDate < acc) return current.startDate;
+    return acc;
+  }, new Date(3000, 0, 1)); // Set Date to Year 3000
+};
 
-  const currentEventChanges = eventChanges.filter((ec) => ec.eventId === task.id);
+export const getProjectEndDate = (project: ProjectPreview): Date => {
+  return project.workPackages.reduce((acc, current) => {
+    if (current.endDate > acc) return current.endDate;
+    return acc;
+  }, new Date(0));
+};
 
-  const changedEvent = { ...task };
-  for (const eventChange of currentEventChanges) {
-    switch (eventChange.type) {
-      case 'change-end-date': {
-        changedEvent.end = eventChange.newEnd;
-        break;
-      }
-      case 'shift-by-days': {
-        changedEvent.start = dayjs(changedEvent.start).add(eventChange.days, 'days').toDate();
-        changedEvent.end = dayjs(changedEvent.end).add(eventChange.days, 'days').toDate();
-        break;
-      }
+export const transformProjectPreviewToProject = (projectPreview: ProjectPreview, team: Team): Project => {
+  return {
+    ...projectPreview,
+    summary: '',
+    budget: 0,
+    links: [],
+    descriptionBullets: [],
+    materials: [],
+    assemblies: [],
+    duration: 0,
+    tasks: [],
+    favoritedBy: [],
+    wbsElementId: '-1',
+    teams: [team],
+    changes: [],
+    dateCreated: new Date()
+  };
+};
+
+export const transformGanttTaskToWorkPackage = (task: GanttTask): WorkPackage => {
+  return {
+    id: task.id,
+    wbsElementId: task.id,
+    wbsNum: {
+      carNumber: task.carNumber,
+      projectNumber: task.projectNumber,
+      workPackageNumber: task.workPackageNumber
+    },
+    dateCreated: new Date(),
+    name: task.name,
+    orderInProject: task.workPackageNumber,
+    startDate: task.start,
+    endDate: task.end,
+    descriptionBullets: [],
+    blockedBy: task.blockedByIds.map(validateWBS),
+    manager: task.manager,
+    lead: task.lead,
+    status: WbsElementStatus.Active,
+    stage: task.stage,
+    links: [],
+    materials: [],
+    assemblies: [],
+    changes: [],
+    immediatelyBlocking: task.blocking,
+    projectName: '',
+    duration: task.end.getTime() - task.start.getTime() / 1000 / 60 / 60 / 24 / 7
+  };
+};
+
+const applyChangesToBlockedBy = (
```

Comment:
can u add a nice comment to this (probably most of the stuff in this file should have nice comments)